src/App.tsx

```diff
@@ -14,11 +15,127 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+  isReversed: boolean,
+  sortBy: string,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  start = () => {
+    this.setState({ isVisible: true });
+  };
+
+  stop = () => {
+    this.setState({ isVisible: false });
+  };
+
+  sortByAlfabeth = () => {
+    this.setState({
+      sortBy: 'alphabet',
+    });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  };
+
+  render() {
+    const {
+      goods,
+      isVisible,
+      isReversed,
+      sortBy,
+    } = this.state;
+    const visibleGoods = [...goods];
+
+    if (sortBy === 'length') {
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+    }
+
+    if (sortBy === 'alphabet') {
+      visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
```

Comment:
sort and reverse are mutable methods so you need to make copy before using these methods